src/hazelcore/ZTyp.re

```diff
@@ -74,10 +76,11 @@ and is_after_zopseq = zopseq => ZOpSeq.is_after(~is_after_zoperand, zopseq)
 and is_after_zoperand =
   fun
   | CursorT(cursor, Hole)
-  | CursorT(cursor, Unit)
-  | CursorT(cursor, Int)
-  | CursorT(cursor, Float)
-  | CursorT(cursor, Bool) => cursor == OnDelim(0, After)
+  | CursorT(cursor, Unit) => cursor == OnDelim(0, After)
+  | CursorT(cursor, Int) => cursor == OnText(3)
```

Comment:
instead of magic numbers here, define constants determined by calling length on the string representation of each type (based on to_string)